### PR TITLE
Add Zandronum egg

### PR DESCRIPTION
**Pull request by Frpst12**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
### New Server Submissions:

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your server use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

### CODE REVIEW

**Comment by Frpst12** [comment]
**SORRY FOR SO MANY COMMITS IS MY FIRST TIME DOING GITHUB******

File: game_eggs/doom/zandronum/egg-zandronum.json
Comment (gOOvER) [inline game_eggs/doom/zandronum/egg-zandronum.json:13]: Source is desined for source games. You can use  `ghcr.io/parkervcp/yolks:debian` insteed
`

**Comment by gOOvER** [comment]
`if [ "$(uname -m)" = x86_64 ]; then
    z="zandronum"$version"-linux-x86_64.tar.bz2"
else
    z="zandronum"$version"-linux-x86.tar.bz2"
fi `

i think, you can use only `x86_64`

**Comment by Frpst12** [comment]
> `if [ "$(uname -m)" = x86_64 ]; then z="zandronum"$version"-linux-x86_64.tar.bz2" else z="zandronum"$version"-linux-x86.tar.bz2" fi `
> 
> i think, you can use only `x86_64`

Sorry, overlooked it when going off of the official wiki, should be resolved now

**Comment by gOOvER** [comment]
Some more things:

`unzip -j freedoom.zip '*/*'` -> you missed the `-o` flag. Install stucks, when server is allready installed


mb you can create these files with touch
```
IPFileParser::parseIPList: could not open adminlist.txt: No such file or directory
IPFileParser::parseIPList: could not open whitelist.txt: No such file or directory
IPFileParser::parseIPList: could not open banlist.txt: No such file or directory
```

**Comment by Frpst12** [comment]
> Some more things:
> 
> `unzip -j freedoom.zip '*/*'` -> you missed the `-o` flag. Install stucks, when server is allready installed
> 
> mb you can create these files with touch
> 
> ```
> IPFileParser::parseIPList: could not open adminlist.txt: No such file or directory
> IPFileParser::parseIPList: could not open whitelist.txt: No such file or directory
> IPFileParser::parseIPList: could not open banlist.txt: No such file or directory
> ```

Thanks for the corrections, added

**Review by Software-Noob — DISMISSED** [review]
Thanks for your contribution :1st_place_medal:. I have a few requests though:

Please extract the example config and upload it to Github as a file into the same folder with the egg. Download the file directly from Github in the egg, thus allowing us to update the config at any time, without forcing users to get a new egg.

The URL to download from would be `https://raw.githubusercontent.com/parkervcp/eggs/master/game_eggs/doom/zandronum/zandronum.ini` once this is merged.

When creating new folders, use `mkdir -p` to prevent attempts of creating folders and throwing errors should they exist, as users will reinstall to update. 

Also, please add this egg to the game_eggs readme.md file.

**Comment by QuintenQVD0** [comment]
The game has prebuild bin for arm64 so please support them.

I already made the install script for you that supports arm64:

```bash
#!/bin/bash
mkdir -p /mnt/server
cd /mnt/server

ARCH=$([[ "$(uname -m)" == "x86_64" ]] && echo "linux-x86_64" || echo "linuxserver-armv8")

echo "Fetching latest version number of Zandronum"
version="$(wget -q -O - http://zandronum.com/download |
sed -n 's/.*The latest version is <strong>\(.*\)<\/strong>.*/\1/p')"

z="zandronum"$version"-${ARCH}.tar.bz2"

echo "Downloading release $version of Zandronum."
echo "Executing wget -nc https://zandronum.com/downloads/$z"
wget -nc https://zandronum.com/downloads/"$z"
echo "Untar Zandronum"
echo "Executing tar -xjvf $z"
tar -xjvf "$z"

echo "Remove useless files"
rm "$z"
rm zandronum # This is the client, we don't need that.

echo "Making zandronum-server executable"
chmod +x zandronum-server

# Create IWAD and PWAD folder
mkdir -p IWAD
mkdir -p PWAD

# Download latest freedoom for the default iwad
freedoom=$(curl -s https://api.github.com/repos/freedoom/freedoom/releases/latest | jq .assets | jq -r .[].browser_download_url | grep -m1 -i freedoom-.*.zip)

echo "$freedoom"
curl -L -o freedoom.zip $freedoom

unzip -j -o freedoom.zip '*/*' -d IWAD/

# Removing useless files
rm freedoom.zip
rm IWAD/*.html
rm IWAD/*.pdf

# Download the configuration file. The server fail to generate a config by itself so we have to supply one or it will crash
wget https://raw.githubusercontent.com/parkervcp/eggs/master/game_eggs/doom/zandronum/zandronum.ini -O /mnt/server/zandronum.ini

touch adminlist.txt
touch whitelist.txt
touch banlist.txt

echo "Install finished."
```

On startup I get:
```logs
[Pterodactyl Daemon]: Finished pulling Docker container image
:/home/container$ ./zandronum-server -port ${SERVER_PORT} -config /home/container/zandronum.ini +set dmflags ${DMFLAGS} +set dmflags2 ${DMFLAGS2} +set zadmflags ${ZADMFLAGS} +set compatflags ${COMPATFLAGS} +set compatflags2 ${COMPATFLAGS2} +set zacompatflags ${ZACOMPATFLAGS} -iwad IWAD/${IWAD} -file PWAD/${PWADS} -host ${MAX_PLAYERS} +map ${MAP} -skill ${SKILL} -netmode 1
Zandronum 3.1 - 211211-2135 - SDL version
Compiled on Dec 12 2021
M_LoadDefaults: Load system defaults.
Gameinfo scan took 0 ms
W_Init: Init WADfiles.
 adding /home/container/zandronum.pk3, 696 lumps
 adding IWAD/freedoom1.wad, 3081 lumps
 adding PWAD/, 0 lumps
usage: set <variable> <value>
usage: set <variable> <value>
usage: set <variable> <value>
usage: set <variable> <value>
usage: set <variable> <value>
usage: set <variable> <value>
I_Init: Setting up machine state.
CPU Vendor ID: GenuineIntel
  Name: Intel(R) Core(TM) i7-3632QM CPU @ 2.20GHz
  Family 6, Model 58, Stepping 9
  Features: MMX SSE SSE2 SSE3 SSSE3 SSE4.1 SSE4.2
Using in-memory database. The database will not be saved on exit.
S_Init: Setting up sound.
ST_Init: Init startup screen.
Checking cmd-line parameters...
S_InitData: Load sound definitions.
G_ParseMapInfo: Load map definitions.
Texman.Init: Init texture manager.
ParseTeamInfo: Load team definitions.
LoadActors: Load actor definitions.
R_Init: Init Doom refresh subsystem.
DecalLibrary: Load decals.
Adding dehacked patch freedoom1.wad:DEHACKED
Patch installed
PWO_Init: Initializing preferred weapon order.
M_Init: Init menus.
P_Init: Init Playloop state.
ParseSBarInfo: Loading default status bar definition.
ParseSBarInfo: Loading custom status bar definition.
===========================================================================
This is Freedoom, the free content first person shooter.
Freedoom is freely redistributable under the terms of the modified BSD
license. Check out the Freedoom website for more information:
    https://freedoom.github.io/
============================================================================
D_CheckNetGame: Checking network game status.
using alternate port 25567
player 1 of 1 (1 nodes)
Initializing network subsystem.
Server using alternate port 25567.
IP address 172.18.0.2:25567
Error Opening file ./GeoIP.dat
GeoIP initialization failed.
UDP Initialized.
container@pterodactyl~ Server marked as running...
*** e1m1: C1M1: Outer Prison ***
```

Is the `Error Opening file ./GeoIP.dat
GeoIP initialization failed.` normal?

And please resolve the 2 confilcts and update it to support arm64 then this can be merged if it is joinable

**Review by QuintenQVD0 — APPROVED** [review]

**Comment by QuintenQVD0** [comment]
LGTM thx!

**Pull request merged**
